api/src/Page/Sample.php

```diff
@@ -1928,8 +1925,11 @@ function _update_sample()
             $dcps = $this->db->pq("SELECT dataCollectionPlanId FROM BLSample_has_DataCollectionPlan
                                 WHERE blSampleId = :1", array($this->arg('sid')));
 
-            if (sizeof($dcps)) {
-                foreach ($dcps as $dcp) {
+            if(sizeof($dcps)) {
+                $maxLocation = $this->_get_current_max_dcp_plan_order($this->args['CONTAINERID']);
```

Comment:
```suggestion
```
Is this line needed?

Reply:
No but the line `$maxLocation = $maxLocation ? $maxLocation + 1 : 0;` but I didn't like it. So instead I have moved all the code for determining if there is alread an plan order into that function and called it get next index.
Could you review and I will put it on ipsyb-dev-5